client/src/controllers/DrilldownController.test.js

```diff
@@ -0,0 +1,214 @@
+                            </div>
+                        </button>
+                    </div>
+
+                    <div
+                        className="w-drilldown__submenu"
+                        id="drilldown-field-0"
+                        hidden
+                        tabIndex="-1"
+                    >
+                        <button
+                            className="w-drilldown__back"
+                            type="button"
+                            aria-label="Back"
+                            data-action="click->w-drilldown#close"
+                        >
+                            Back
+                        </button>
+                    </div>
+                </div>
+            </div>
+            <button id="action-element" data-controller="w-action">Action</button>
+        </section>`,
+    identifier = 'w-drilldown',
+  ) => {
+    document.body.innerHTML = `<main>${html}</main>`;
+
+    application = new Application();
+    application.handleError = (error, message) => {
+      errors.push({ error, message });
+    };
+    application.register(identifier, DrilldownController);
+    application.register('w-action', ActionController);
+    application.register('w-dropdown', DropdownController);
+    application.start();
+
+    await jest.runAllTimersAsync();
+    controller = application.getControllerForElementAndIdentifier(
+      document.querySelector(`[data-controller="${identifier}"]`),
+      identifier,
+    );
+
+    actionElement = document.getElementById('action-element');
+
+    return [
+      ...document.querySelectorAll(`[data-controller~="${identifier}"]`),
+    ].map((element) =>
+      application.getControllerForElementAndIdentifier(element, identifier),
+    );
+  };
+
+  beforeEach(async () => {
+    await setup();
+  });
+
+  afterEach(() => {
+    application?.stop && application.stop();
+    errors = [];
+    eventNames.forEach((name) => {
+      events[name] = [];
+    });
+  });
+
+  describe('DrilldownController open/close actions', () => {
+    it('opens the submenu and sets active submenu value when toggle is clicked', async () => {
+      const toggleButton = document.querySelector(
+        '[data-w-drilldown-target="toggle"]',
+      );
+      const submenu = document.querySelector('#drilldown-field-0');
+
+      toggleButton.click();
+      await jest.runAllTimersAsync();
+
+      expect(controller.activeSubmenuValue).toBe('drilldown-field-0');
+      expect(toggleButton.getAttribute('aria-expanded')).toBe('true');
+      expect(submenu.hidden).toBeFalsy();
+    });
+
+    it('closes the submenu and clears active submenu value when close action is triggered', () => {
+      const toggleButton = document.querySelector(
+        '[data-w-drilldown-target="toggle"]',
+      );
+      toggleButton.click();
+
+      controller.close();
+
+      const submenu = document.querySelector('#drilldown-field-0');
+      expect(controller.activeSubmenuValue).toBe('');
```

Comment:
Try to avoid reading values directly from the controller.

Either test the desired behaviour (as in no sub-menu, the previous one being closed)

OR if really needed, you can check the attribute in the DOM `data-w-drilldown-active-submenu-value`.

Here, though it looks like you do not need this line here as you already are checking the DOM in the lines below.